core/src/main/java/nu/yona/server/messaging/service/FirebaseService.java

```diff
@@ -78,19 +84,11 @@ public void sendMessage(String registrationToken, nu.yona.server.messaging.entit
 		Message firebaseMessage = Message.builder().setNotification(new Notification(title, body))
 				.putData("messageId", Long.toString(getMessageId(message))).setToken(registrationToken).build();
 
-		if (yonaProperties.getFirebase().isEnabled())
-		{
-			logger.info("Sending Firebase message");
-			// Sending takes quite a bit of time, so do it asynchronously
-			CompletableFuture.runAsync(() -> sendMessage(firebaseMessage))
-					.whenCompleteAsync((r, t) -> logIfCompletedWithException(t));
-		}
-		else
-		{
-			logger.info("Firebase message not sent because Firebase is disabled");
-			// Store for testability
-			lastMessageByRegistrationToken.put(registrationToken, firebaseMessage);
-		}
+		// Sending takes quite a bit of time, so do it asynchronously
+		ThreadData threadData = asyncExecutor.getThreadData();
+		CompletableFuture
+				.runAsync(() -> asyncExecutor.initThreadAndDo(threadData, () -> sendMessage(registrationToken, firebaseMessage)))
+				.whenCompleteAsync((r, t) -> logIfCompletedWithException(t, registrationToken));
```

Comment:
Does this `logIfCompletedWithException` not need an `initThreadAndDo` call?

Reply:
I don't think so. I believe I have manually verified this. If it is lacking, we'll need to add this in a separate PR.